packages/koa-metrics/src/timing.ts

```diff
@@ -0,0 +1,29 @@
+import {StatsD} from 'hot-shots';
+import {Context} from 'koa';
+
+export interface Timer {
+  close(responseClient?: StatsD): void;
+}
+
+export function initTimer(client: StatsD, ctx: Context): Timer {
```

Comment:
While this factory is pretty nice compared to before, I wonder if there are a couple separate concerns here? 
- a generic `Timer` that tells us the high res time in milliseconds
- getting the queue time from the header
- sending the queue time to statsD 
- sending the end time to statsD 

I picture something a bit simpler than this that has the same api

```ts
  () => {
    stop(): number
  }
```

and then the logic to send the request start and end as a separate call

and then you would do

```ts
const requestQueueTime = ctx.request.get(Headers.QueueTime);
client.timing(Tags.QueueTime, requestQueueTime);

const timer = initTimer()

// ...rest of thing

const duration = timer.stop();
responseClient.timing(Tags.EndTime, duration)
```

I feel like this split makes things a bit clearer (and means once we have our own logger that supports timers we can pull timer out and use that).

This would also make it easier for other things in the app to pull in the timer logic and use it to time whatever they care about.